loader/dd_library_loader.c

```diff
@@ -0,0 +1,355 @@
+    }
+
+    // The code below basically comes from the function "php_load_extension" in "ext/standard/dl.c",
+    // which does not allow loading an extension using a full path.
+
+    zend_module_entry *module_entry;
+    zend_module_entry *(*get_module)(void);
+
+    LOG(INFO, "Found extension file: %s", ext_path);
+
+    void *handle = DL_LOAD(ext_path);
+    if (!handle) {
+        LOG(ERROR, "Cannot load the extension");
+        goto abort;
+    }
+
+    get_module = (zend_module_entry * (*)(void)) ddloader_dl_fetch_symbol(handle, "_get_module");
+    if (!get_module) {
+        LOG(ERROR, "Cannot fetch the module entry");
+        goto abort_and_unload;
+    }
+
+    module_entry = get_module();
+
+    if (module_entry->zend_api != php_api_no) {
+        LOG(ERROR, "API number mismatch between module (%d) and runtime (%d)", module_entry->zend_api, php_api_no);
+        goto abort_and_unload;
+    }
+    if (strcmp(module_entry->build_id, module_build_id)) {
+        LOG(ERROR, "Build ID mismatch between module (%s) and runtime (%s)", module_entry->build_id, module_build_id);
+        goto abort_and_unload;
+    }
+
+    /**
+     * At that point, we don't know if ddtrace will be registered or not by the PHP configuration.
+     * So we register it under the name "ddtrace_injected", add set an optional dependency to
+     * "ddtrace" to be sure that our injected extension will be started up after the regular "ddtrace"
+     * (if it's loaded!), and finally we wrap the MINIT function to perform our checks there.
+     */
+    module_entry->name = "ddtrace_injected";
+    origin_ddtrace_module_startup_func = module_entry->module_startup_func;
+    module_entry->module_startup_func = ZEND_MODULE_STARTUP_N(ddtrace_injected);
+    // Set our optional dependency to the real "ddtrace" module
+    orig_ddtrace_module_deps = module_entry->deps;
+    module_entry->deps = ddtrace_injected_module_deps;
+    // Backup the function list and set it to NULL to make sure we don't register the functions twice
+    // They'll be restored if ddtrace is not already registered.
+    orig_functions = module_entry->functions;
+    module_entry->functions = NULL;
+
+    // Register the module, catching all errors that can happen (already loaded, unsatisied dep, ...)
+    void (*old_error_handler)(int, zend_string *, const uint32_t, zend_string *);
+    old_error_handler = zend_error_cb;
+    zend_error_cb = ddloader_error_handler;
+    module_entry = zend_register_internal_module(module_entry);
+    zend_error_cb = old_error_handler;
+
+    if (module_entry == NULL) {
+        LOG(ERROR, "Cannot register the module");
+        goto abort_and_unload;
+    }
+
+    return SUCCESS;
+
+abort_and_unload:
+    LOG(INFO, "Unloading the library");
+    DL_UNLOAD(handle);
+abort:
+    LOG(INFO, "Abort the loader");
+    free(ext_path);
+
+    return SUCCESS;
+}
+
+static inline void ddloader_configure() {
+    if (getenv("DD_LOADER_ENABLE_LOGS")) {
```

Comment:
```suggestion
    if (getenv("DD_TRACE_DEBUG")) {
```
as per the Guardrails RFC